### PR TITLE
Bump golang to 1.21 GA

### PR DESCRIPTION
**Pull request by arukiidou**
Bump golang to 1.21 GA

- test passed https://github.com/arukiidou/go-jose/pull/1

### CODE REVIEW

**Comment by arukiidou** [comment]
@aarongable please take a look

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
         GO_VERSION:
           - "1.16.x"
           - "1.20.x"
-          - "1.21.0-rc.2"
+          - "1.21.x"
```
Comment (aarongable) [inline .github/workflows/go.yml:17]: Go ahead and remove go 1.20.x both here and below. We were only testing both 1.20 and 1.21rc to check that updating to go1.21 wouldn't break anything. Now that it's time to do the update for real, we can leave go 1.20 behind.

**Review by aarongable — COMMENTED** [review]
Generally LGTM, but see comment.

**Review by aarongable — APPROVED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (arukiidou) [reply .github/workflows/go.yml:16]: @aarongable 
fixed ,and test passed
https://github.com/go-jose/go-jose/pull/54/commits/94a19933299e56103245c6d76cc462d94a4a0475

**Pull request merged**